kernel/rtlil.cc

```diff
@@ -2312,6 +2323,32 @@ RTLIL::SigSpec RTLIL::Module::Initstate(RTLIL::IdString name, const std::string
 	return sig;
 }
 
+RTLIL::ParameterInfo::ParameterInfo()
+{
+	static unsigned int hashidx_count = 123456789;
+	hashidx_count = mkhash_xorshift(hashidx_count);
+	hashidx_ = hashidx_count;
+
+#ifdef WITH_PYTHON
+	RTLIL::ParameterInfo::get_all_parameterinfos()->insert(std::pair<unsigned int, RTLIL::ParameterInfo*>(hashidx_, this));
```

Comment:
This is a global, but has no locking.  Is this safe and/or advisable?

Reply:
That is how all other objects are stored for the Python wrapper. Therefore I believe it is correct.